src/NHibernate/Util/LinkedHashMap.cs

```diff
@@ -182,12 +182,22 @@ private void OverrideEntry(Entry e, TValue value)
 			InsertEntry(e);
 		}
 
-		public virtual ICollection<TKey> Keys
+		public KeyCollection Keys
```

Comment:
Such changes are binary breaking, the class and property being public. So, such a change could only be merged for a new major version. For a faster release of the change, you need to ensure all changes are non-breaking. So, you would have to introduce new properties/methods and obsolete the old ones, instead of changing in a breaking way existing ones.

Reply:
According to your https://github.com/nhibernate/nhibernate-core/issues/2407#issuecomment-636494042

> NHibernate is not meant to be an utility library. This kind of classes should have been made internal in my opinion. In other words, we do not support external uses of LinkedHashMap.

So, maybe, it is even better to make `LinkedHashMap` class internal.

Reply:
I cannot use `ObsoleteAttribute` here, because a class cannot have 2 properties with the same name.

Reply:
> it is even better to make LinkedHashMap class internal

Yes it is better, but that would be a breaking change, so, it would have to wait for a major release.

> I don't care about getting these changes faster.

Unless we start issuing major versions for rather minor changes, I do not think a major version will come any time soon. There are many things that should be done for an actual new major version, but I currently do not see enough contributing workforce available to do them. There are approved PR in the next major milestone lingering there since years...

> I cannot use ObsoleteAttribute here, because a class cannot have 2 properties with the same name.

But you can add it with another name. Granted, that would not be ideal.

I know `GetEnumerator` has a special treatment by the compiler for foreach statement, so for this one, using another name is not an option.